_posts/news/2020-11-16-announcing-scalajs-1.3.1.md

```diff
@@ -0,0 +1,111 @@
+* It is backward source compatible with 1.3.0: source code that used to compile with 1.3.0 should compile as is when upgrading to 1.3.1.
+
+From Scala.js 1.3.0, which was a **minor** release:
+
+* It is *not* forward binary compatible with 1.2.x: libraries compiled with 1.3.1 cannot be used with 1.2.x or earlier.
+* It is *not* entirely backward source compatible: it is not guaranteed that a codebase will compile *as is* when upgrading from 1.2.x or earlier (in particular in the presence of `-Xfatal-warnings`).
+
+As a reminder, libraries compiled with 0.6.x cannot be used with Scala.js 1.x; they must be republished with 1.x first.
+
+## Performance fix on Firefox
+
+Scala.js code has been known for bad performance on Firefox with the default output mode of ECMAScript 2015.
+
+Long story short: this release fixes this issue.
+If you have been using the following setting to avoid bad performance on Firefox, you can now remove it:
+
+{% highlight scala %}
+// This was necessary before to avoid bad performance on Firefox
+scalaJSLinkerConfig ~= { _.withESFeatures(_.withUseECMAScript2015(false)) }
+{% endhighlight %}
+
+For the longer story: it turns out that SpiderMonkey, the JavaScript engine of Firefox, exhibits severe performance penalties for JavaScript `class`es.
+This has caused slowdowns for Scala.js code of up to 10x compared to other browsers and compared to the ECMAScript 5.1 mode.
+Starting from Scala.js 1.3.1, we restrict the usage of `class`es in the emitted JavaScript to the minimum required for correct semantics.
+We fall back to `function`s and `prototype`s for the majority of the emitted code.
+
+This change has a cost in terms of emitted code size.
+If your codebase does not target the Web nor any SpiderMonkey-based environment, you can tell Scala.js that avoiding `class`es is not necessary with the following setting:
+
+{% highlight scala %}
+// Trade away decent Firefox performance for a smaller output
+scalaJSLinkerConfig ~= { _.withESFeatures(_.withAvoidClasses(false)) }
+{% endhighlight %}
+
+## Miscellaneous
+
+### Better support of Scala 3 in sbt-scalajs
+
+The sbt plugin now has better support for Scala 3 codebases.
+In particular, the JUnit support works out of the box with `enablePlugins(ScalaJSJUnitPlugin)`.
+It is also future-proof for when sbt will support Scala 3 out of the box, rather than requiring `sbt-dotty`.
+
+These improvements are only available for Scala >= 3.0.0-M1.
+The 0.x versions are not supported.
+
+### New JDK APIs
+
+This release contains some improvements in the JDK APIs that we support:
+
+New classes:
+
+* `java.lang.ClassValue`
+
+New methods:
+
+* `new java.util.HashMap(initialCapacity: Int, loadFactor: Float)`
+* `java.util.Objects.requireNonNull(obj: T, message: Supplier[String])`
+
+Methods with fixed or improved behavior:
+
+* In `java.util.regex.Matcher`:
+  * `lookingAt()` and `matches()` ([issue #4252](https://github.com/scala-js/scala-js/issues/4252))
```

Comment:
consider dropping `issue` in the text (also below, further, note the inconsistent linkage).